dot/network/block_announce.go

```diff
@@ -138,20 +138,14 @@ func (*BlockAnnounceHandshake) Type() byte {
 	return 0
 }
 
-// Hash returns blake2b hash of block announce handshake.
-func (hs *BlockAnnounceHandshake) Hash() (common.Hash, error) {
-	// scale encode each extrinsic
-	encMsg, err := hs.Encode()
-	if err != nil {
-		return common.Hash{}, fmt.Errorf("cannot encode handshake: %w", err)
+// IsHandshake returns true if handshakes's role is valid.
+func (hs *BlockAnnounceHandshake) IsHandshake() bool {
```

Comment:
Since `BlockAnnounceHandshake` is named with `Handshake`, shouldn't it always be a handshake?

Reply:
Renaming this function to `IsValidHandshake` to make things more clear